apps/openchallenges/edam-etl/src/main.py

```diff
@@ -1,16 +1,100 @@
-"""Extract, transform and load EDAM concepts"""
-
+import pandas as pd
+import requests
 from os import getenv
+from typing import Optional
+import regex as re
 
 # Get config from the environment variables
-oc_db_url = getenv("OC_DB_URL")
-print(f"OC DB URL: {oc_db_url}")
 
-# TODO Download EDAM concepts from GitHub or S3 bucket (CSV file)
-print("[TODO] Download EDAM concepts from GitHub or S3 bucket (CSV file)")
+OC_DB_URL = getenv("OC_DB_URL")
+VERSION = getenv("VERSION")
+print(f"EDAM Version: {VERSION}")
+print(f"OC DB URL: {OC_DB_URL}")
+
+
+def download_edam_csv(url: str, version: str) -> Optional[bool]:
+    """Download EDAM concepts from GitHub or S3 bucket (CSV file)"""
+    print("Downloading the EDAM concepts from GitHub (CSV file)...")
+    try:
+        response = requests.get(url)
+        response.raise_for_status()  # Raise an exception for bad response
+        with open(f"EDAM_{version}.csv", "wb") as f:
+            f.write(response.content)
+        print("EDAM concepts downloaded successfully.")
+        return True
+    except requests.RequestException as e:
+        print(f"Error downloading EDAM concepts: {e}")
+        return None
```

Comment:
This doesn't matter, but I'm just curious: why return `None` instead of `False`?

Reply:
False indicates a failure, i.e. the download failed because the operation didn't do what it was supposed to do properly and there is a boolean checking to see if the operation fails, while None just says that there is no boolean since there's no meaningful output. Sonar suggested that I add an exception many pushes to this pr ago so I just added that since IMO the use of an exception wasn't really necessary since we know that either you're getting a download when executing the function or not. My original function had no exceptions until the Sonar Issue was flagged.